src/test/java/com/google/javascript/cl2dts/nested_enums.js

```diff
@@ -0,0 +1,24 @@
+goog.provide('nested.bar.Baz');
+goog.provide('nested.bar.Baz.NestedEnum');
+goog.provide('nested.bar.SuperBaz');
+goog.provide('nested.bar.SuperBaz.NestedEnum');
+
+/** @constructor */
+nested.bar.Baz = function() {};
+
+/** @enum */
+nested.bar.Baz.NestedEnum = {
+  A: 1
+};
+
+/**
+ * @constructor
+ * @extends {nested.bar.Baz}
+ */
+nested.bar.SuperBaz = function() {};
+
+/** @enum */
+nested.bar.SuperBaz.NestedEnum = nested.bar.Baz.NestedEnum;
```

Comment:
I suppose there is also the case of aliasing enum from a source that isn't a supertype, but you naturally handle that as well?